src/SlackConnector/Connections/Clients/Chat/FlurlChatClient.cs

```diff
@@ -13,13 +13,14 @@ internal class FlurlChatClient : IChatClient
     {
         private readonly IResponseVerifier _responseVerifier;
         internal const string SEND_MESSAGE_PATH = "/api/chat.postMessage";
+        internal const string DELETE_MESSAGE_PATH = "/api/chat.delete";
 
         public FlurlChatClient(IResponseVerifier responseVerifier)
         {
             _responseVerifier = responseVerifier;
         }
 
-        public async Task PostMessage(string slackKey, string channel, string text, IList<SlackAttachment> attachments)
+        public async Task<PostMessageResponse> PostMessage(string slackKey, string channel, string text, IList<SlackAttachment> attachments)
```

Comment:
Okay, I chatted with a colleague about this because I thought I was missing something.

The outcome was that it seems like you're doing extra testing work for not much gain?

For example:
```
        public async Task<SlackMessagePosted> Say(BotMessage message)
        {
            if (string.IsNullOrEmpty(message.ChatHub?.Id))
            {
                throw new MissingChannelException("When calling the Say() method, the message parameter must have its ChatHub property set.");
            }

            var client = _connectionFactory.CreateChatClient();
            var response = await client.PostMessage(SlackKey, message.ChatHub.Id, message.Text, message.Attachments);
            if (response == null)
            {
                return null;
            }
            return new SlackMessagePosted() { TimeStamp = response.TimeStamp, Channel = response.Channel, Message = response.Message };
        }
```

You're verifying this call with a test of
```
            [Test]
            public void then_should_call_messenger()
            {
                GetMockFor<IChatClient>()
                    .Verify(x => x.PostMessage(SlackKey, Message.ChatHub.Id, Message.Text, Message.Attachments), Times.Once);
            }
```

So basically you're testing that the Say call is passing the expected values to the PostMessage function within the Say function.

But considering that the Say function's primary purpose is to call that function and the compiler can already verify that you're passing the correct information at compile time (albeit possibly not if you have the same argument types but in the wrong order), it seems like wasted effort. 

If I now expand that test to check for the return type, I'm again just shadowing what the compiler could do for me.